app/services/profanity_detector.rb

```diff
@@ -0,0 +1,56 @@
+# Detects profanity in a string, the list of profane words
+# comes from the profanity_filter gem for now
+module ProfanityDetector
+  module_function
+
+  # Calls a block until the result does not contain profanity
+  # @yield block that generates a random value
+  # @yieldreturn [String] string to check for profanity
+  def without_profanity
+    loop do
+      word = yield
+
+      return word unless profane?(word)
+    end
+  end
+
+  # Returns true if the string contains profanity inside it
+  # ProfanityFilter::Base.profane? splits by word, but this
+  # checks all substrings
+  def profane?(str)
+    preprocess_if_needed!
+
+    str_chars = str.gsub(/\W/, '').downcase.chars
+
+    (min_profanity_length..[str.length, max_profanity_length].min).each do |size|
```

Comment:
Since we're only wanting to consider the filtered value, should this be...

```suggestion
    (min_profanity_length..[str_chars.length, max_profanity_length].min).each do |size|
```

?